### DISCUSSION TITLE
Should always be only one drag-bar per stereo track

### DISCUSSION THREAD

**Discussion by petersampsonaudacity** [audacity/audacity#1484] [Development]
This discussion is mainly aimed @Tantacrul and @jounih - as the two UX/GUI experts on the Muse team.

From previous discussion I know that it is planned just to have a single drag-bar affordance for each stereo tack - which makes sense as both channels are then ganged and dragged together.

There was also a discussion that suggested the when a stereo track was formed from two single channel tracks (split-stereo or mono) that in this case the right channel should have its own drag bar - i.e. two drag bars in the stereo track.

I think this is a very bad idea unless and until the user is then able to drag each channel of a stereo track, or clips within each channel of a stereo track, independently.  I, as a user, if I saw two different drag bars one per channel would realistically assume that I could drag each one separately - _that is not an unrealistic expectation_.

This seems on the surface an interesting and somewhat beguiling idea but:
a) I suspect it would an awful lot of coding work.
b) is really unnecessary as the user who want to to that already has a **_simple_** way of achieving that.

Steps:
1) get a stereo track (with some clips ideally)
2) Split Stereo Track
3) Observe: the left channel is panned hard left and the right channel is panned hard right
4) Press play
5) Observe: the output audio sounds the same as if you had played the stereo track _(I use Sgt Pepper as a good source with well-differentiated left and right channels)_
6) move clips around independently in each channel
7) Play to review the result
8) Make Stereo track

**Comment by JamesCrook** [comment]
...which in turn leads to the question of whether one even should allow forming a stereo track from two mismatched mono tracks.  We do right now, and I think we should continue to.  And then, with drag bars in the mix, it would be logical to give each channel its own drag bars rather than privilege one channel over the other...

It feels very edge-case to go and support stereo tracks with independent dragging within channels.  It's already edge case to support merging mismatched tracks to stereo.

I think the drag bar in each channel suggestion arose from a desire for 'make stereo' to remain reversible.  The argument was that padding with silence changed the clips, so that 'split stereo' would no longer undo 'make stereo'.

I'd say 'make stereo' is reversible, by 'undo'.  It's OK, and probably good, to pad the mismatches with silence.  'Split Stereo' will then no longer be an inverse to 'Make Stereo' in all cases, and that's OK.  One then doesn't have to worry about ever having two lots of drag handles in stereo tracks, since the clips will always match up.

**Comment by Paul-Licameli** [comment]
I don’t like where this is leading.  What should have been a simple UX improvement is leading to changes in the data model (either add padding, or make distinctions between stereo tracks based on how they were made stereo rather than the stereo content).

The latest iteration calls for making the distinction which is surprisingly hard to do right in view of the problems of upgrade or downgrade of the file format.

My opinion remains:  if two channels are merged, with clips that align and could be joined — then let the user choose whether to have right channel drag bars!  Add something to the drop down menu.  Just don’t impose a choice that can’t be reversed.  It is the attempt to do that, that is leading to so much other debate and complication.

**Reply by petersampsonaudacity** [reply]
> What should have been a simple UX improvement is leading to changes in the data model (either add padding, 
>or make distinctions between stereo tracks based on how they were made stereo rather than the stereo content).

The silence "padding" was Jame's additional suggestion not mine.

My suggestion remains:
a) There should only be one type of stereo track
b) and should be with a single drag-bar affordance
c) there should not be a second type of stereo track with one drag-bar per channel (just because it is formed from two mono tracks 

This provides purity and understand-ability of UX/GUI - and should not involve changes to any file format.

If you really want to remain with a special stereo track that has one drag-bar per channel then really, as I stated in my first post in this discussion thread, then you do really have to allow the user to drag each channel separately - and that would be likely to involve changes to file format and certainly would be additional coding.

Consider also the use case of a more complex multi-channel setup rather than simple two-channel stereo.  There are now soundcards capable of handling six-channel 5.1 audio,  Would you really want a 5.1 track made from 6 mono channels to be carrying six drag-bars, one per channel?  That doesn't look sensible to me.


History has shown us that often with Audacity what looks like a simple fix, change or addition can have complex changes elsewhere in the app - consider say Sync-lock or Macros as a couple of examples.


> My opinion remains: if two channels are merged, with clips that align and could be joined — then let the user choose 
>whether to have right channel drag bars! Add something to the drop down menu. Just don’t impose a choice that can’t 
>be reversed. It is the attempt to do that, that is leading to so much other debate and complication.

I'm failing to see what can't be reversed if a user creates a stereo track from two mono tracks and later decides to split them to mono for further individual editing of either mono track.

As the current proposal stands (assuming I remember correctly from testing - I think it was @vsverchinsky's  test build) when the user creates a stereo track from two mono tracks or two split stereo tacks (or even a mixture of those) then they get a steero track with two drag bars one per channel.  There is no user choice, that's just what they get.

There is a workaround if the user wants a single drag-bar and that is to Mix&Render the stereo track.


But at the end of the day it is not my decision.  I wrote this Issue up as a thought piece for the UX experts, the GUI designers, of Audacity to consider before they pressed ahead.  So I think now it's over to @Tantacrul and @jounih to consider this.

_I'm just glad that if it stays the current proposed way I won't be having to deal with disgruntled users who don't understand why they have two different types of stereo track and/or why they can't drag the stereo channels indecently when they have two drag bars._

**Comment by petersampsonaudacity** [comment]
Consider this use case - testing Vitaly's build on W10 with: audacity win-3.0.4-alpha-20210810+cae0f9d-64bit

1) get a stereo track
![image](https://user-images.githubusercontent.com/50205984/129731141-7bba2abe-bcb7-48e4-8837-ca2eb2ca833f.png)
Observe one drag bar only for the stereo track

2) Split the stereo track
![image](https://user-images.githubusercontent.com/50205984/129731230-1262903d-3567-499a-a56e-59d3bf988517.png)
Observe one drag bar per track


3) Make Stereo track
![image](https://user-images.githubusercontent.com/50205984/129731485-02707f4a-00c5-4866-bc4b-5253cffe692a.png)
Observe two drag bars in the one track one per channel - thus different from Step 1


4) Mix & Render the stereo track from step 3
![image](https://user-images.githubusercontent.com/50205984/129732051-7f7ec846-0a77-4320-86fc-27e99f2d6029.png)
Observe:  the stereo track now only has one drag-bar again.


I find it most odd that at step 3 the same stereo track is different as to how it is in steps 1 and 4 - ad I suspect that many users will feel the same.
